### PR TITLE
Update delete_files() helper function

### LINKED ISSUE #3015: Bug: delete_files() and hidden directories
**Describe the bug**
`delete_files()` in the Filesystem Helper currently ignores hidden directories. I'm not sure if this is a "bug" since the code seems to do it deliberately (notice the first line):
```
if (is_dir($path . DIRECTORY_SEPARATOR . $filename) && $filename[0] !== '.')
{
	delete_files($path . DIRECTORY_SEPARATOR . $filename, $del_dir, $htdocs, $_level + 1);
}
elseif ($htdocs !== true || ! preg_match('/^(\.htaccess|index\.(html|htm|php)|web\.config)$/i', $filename))
{
	@unlink($path . DIRECTORY_SEPARATOR . $filename);
}
```
But this isn't documented anywhere and seems an odd exclusion. For example, cleaning up an example project `delete_files()` fails to remove the root directory because it contains **.github/**.

If leaving hidden directories is intentional, then I would be in favor of giving this the same treatment as `get_filenames()`: remove the recursive call, use a `RecursiveIteratorIterator` instead, and add a parameter for `bool $hidden = false`.

**CodeIgniter 4 version**
`develop`

**Affected module(s)**
Helpers > Filesystem

**Expected behavior, and steps to reproduce if appropriate**
Given:
ROOT/app/[files]
ROOT/writable/[files]
ROOT/.github/[files]

I would expect calling `delete_files('ROOT', true)` to wipe the whole project, but it does not.

**Context**
 - OS: Linux
 - Web server: Apache
 - PHP version: 7.3

### PR DESCRIPTION
**Pull request by michalsn**
**Description**
Support for removing hidden folders. See: #3015

**Checklist:**
- [x] Securely signed commits
- [ ] Component(s) with PHPdocs
- [x] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

File: system/Helpers/filesystem_helper.php [hunk 1]
```diff
@@ -160,39 +160,48 @@ function write_file(string $path, string $data, string $mode = 'wb'): bool
...
 				}
 			}
 
-			closedir($current_dir);
-
-			return ($del_dir === true && $_level > 0) ? @rmdir($path) : true;
+			return true;
 		}
-		catch (\Exception $fe)
+		catch (\Exception $e)
```
Comment (paulbalandan) [inline system/Helpers/filesystem_helper.php:204]: Just asking. Should we use `\Throwable` here instead?

Comment (michalsn) [reply system/Helpers/filesystem_helper.php:204]: Yes, thank you.

**Comment by michalsn** [comment]
The thing is that we need our files to be in a certain order so we can delete them and `get_filenames()` function doesn't give us that. 

Of course, we can work on the result returned by `get_filenames()` but it won't be very pretty. I would stay with the current approach, even if these two functions are a bit similar.

**Comment by michalsn** [comment]
I'm not sure why Coveralls show decreased test coverage. I will say it's a mistake and merge.

**Comment by MGatner** [comment]
I’ve found the Coveralls CI to be unreliable on small PRs. Not worried about it.

**Pull request merged**